ras_party/controllers/account_controller.py

```diff
@@ -275,23 +274,29 @@ def request_password_change(payload, session):
     if not respondent:
         raise RasError("Respondent does not exist.", status=404)
 
-    email_address = respondent.email_address
-    verification_url = PublicWebsite().reset_password_url(email_address)
+    logger.debug("Requesting password change", party_id=respondent.party_uuid)
 
-    personalisation = {
-        'RESET_PASSWORD_URL': verification_url,
-        'FIRST_NAME': respondent.first_name
-    }
+    if respondent.status is RespondentStatus.ACTIVE:
 
-    logger.info('Reset password url', url=verification_url)
+        email_address = respondent.email_address
+        verification_url = PublicWebsite().reset_password_url(email_address)
 
-    party_id = respondent.party_uuid
-    try:
-        NotifyGateway(current_app.config).request_password_change(
-            email_address, personalisation, str(party_id))
-    except RasNotifyError:
-        # Note: intentionally suppresses exception
-        logger.error('Error sending notification email for party_id', party_id=party_id)
+        personalisation = {
+            'RESET_PASSWORD_URL': verification_url,
+            'FIRST_NAME': respondent.first_name
+        }
+
+        logger.info('Reset password url', url=verification_url, party_id=respondent.party_uuid)
+
+        party_id = respondent.party_uuid
+        try:
+            NotifyGateway(current_app.config).request_password_change(
+                email_address, personalisation, str(party_id))
+        except RasNotifyError:
+            # Note: intentionally suppresses exception
+            logger.error('Error sending notification email for party_id', party_id=party_id)
+
+        logger.debug('Password reset email successfully sent', party_id=respondent.party_uuid)
 
     return {'response': "Ok"}
```

Comment:
Should we still be returning a response ok when we haven't actually done what the call is expecting us to do (send a password change link)?

Reply:
Yes this is as Dan has said it should carry on to the same screen etc just not send the email